bundle/deploy/terraform/convert.go

```diff
@@ -79,222 +78,64 @@ func BundleToTerraformWithDynValue(ctx context.Context, root dyn.Value) (*schema
-		if src.ModifiedStatus == "" && src.ID == "" {
-			src.ModifiedStatus = resources.ModifiedStatusCreated
-		}
-	}
-	for _, src := range config.Resources.RegisteredModels {
-		if src.ModifiedStatus == "" && src.ID == "" {
-			src.ModifiedStatus = resources.ModifiedStatusCreated
-		}
-	}
-	for _, src := range config.Resources.QualityMonitors {
-		if src.ModifiedStatus == "" && src.ID == "" {
-			src.ModifiedStatus = resources.ModifiedStatusCreated
-		}
-	}
-	for _, src := range config.Resources.Schemas {
-		if src.ModifiedStatus == "" && src.ID == "" {
-			src.ModifiedStatus = resources.ModifiedStatusCreated
-		}
-	}
-	for _, src := range config.Resources.Volumes {
-		if src.ModifiedStatus == "" && src.ID == "" {
-			src.ModifiedStatus = resources.ModifiedStatusCreated
-		}
-	}
-	for _, src := range config.Resources.Clusters {
-		if src.ModifiedStatus == "" && src.ID == "" {
-			src.ModifiedStatus = resources.ModifiedStatusCreated
-		}
-	}
-	for _, src := range config.Resources.Dashboards {
-		if src.ModifiedStatus == "" && src.ID == "" {
-			src.ModifiedStatus = resources.ModifiedStatusCreated
-		}
-	}
-	for _, src := range config.Resources.Apps {
-		if src.ModifiedStatus == "" {
-			src.ModifiedStatus = resources.ModifiedStatusCreated
+func TerraformToBundle(ctx context.Context, state ExportedResourcesMap, config *config.Root) error {
+	return config.Mutate(func(v dyn.Value) (dyn.Value, error) {
+		var err error
+		v, err = ensureMap(v, dyn.Path{dyn.Key("resources")})
+		if err != nil {
+			return v, err
 		}
-	}
-	for _, src := range config.Resources.SecretScopes {
-		if src.ModifiedStatus == "" {
-			src.ModifiedStatus = resources.ModifiedStatusCreated
+
+		for groupName, group := range state {
+			var err error
+			v, err = ensureMap(v, dyn.Path{dyn.Key("resources"), dyn.Key(groupName)})
+			if err != nil {
+				return v, err
+			}
+
+			for resourceName, attrs := range group {
+				path := dyn.Path{dyn.Key("resources"), dyn.Key(groupName), dyn.Key(resourceName)}
+				resource, err := dyn.GetByPath(v, path)
+				if !resource.IsValid() {
+					m := dyn.NewMapping()
+					m.SetLoc("id", nil, dyn.V(attrs.ID))
+					m.SetLoc("modified_status", nil, dyn.V(resources.ModifiedStatusDeleted))
+					v, err = dyn.SetByPath(v, path, dyn.V(m))
+					if err != nil {
+						return dyn.InvalidValue, err
+					}
+				} else if err != nil {
+					return dyn.InvalidValue, err
+				} else {
+					v, err = dyn.SetByPath(v, dyn.Path{dyn.Key("resources"), dyn.Key(groupName), dyn.Key(resourceName), dyn.Key("id")}, dyn.V(attrs.ID))
+					if err != nil {
+						return dyn.InvalidValue, err
+					}
+				}
+			}
 		}
-	}
 
-	return nil
+		return dyn.MapByPattern(v, dyn.Pattern{dyn.Key("resources"), dyn.AnyKey(), dyn.AnyKey()}, func(p dyn.Path, inner dyn.Value) (dyn.Value, error) {
+			idPath := dyn.Path{dyn.Key("id")}
+			statusPath := dyn.Path{dyn.Key("modified_status")}
+			id, _ := dyn.GetByPath(inner, idPath)
+			status, _ := dyn.GetByPath(inner, statusPath)
```

Comment:
Can you include a comment here explaining that IDs are set for resources that have already been deployed and that the status is set for resources that were deleted? The remaining resources are yet to be deployed.